test/Benchmarks/Instrumentation/AspNetCoreInstrumentationBenchmarks.cs

```diff
@@ -0,0 +1,114 @@
+// <copyright file="AspNetCoreInstrumentationBenchmarks.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#if !NETFRAMEWORK
+using System.Net.Http;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry;
+using OpenTelemetry.Trace;
+
+/*
+// * Summary *
+
+BenchmarkDotNet=v0.13.1, OS=Windows 10.0.22000
+Intel Core i7-8850H CPU 2.60GHz (Coffee Lake), 1 CPU, 12 logical and 6 physical cores
+.NET SDK=7.0.100-preview.6.22275.1
+  [Host] : .NET 6.0.8 (6.0.822.36306), X64 RyuJIT
+
+Job=InProcess  Toolchain=InProcessEmitToolchain
+
+|                                      Method |     Mean |   Error |  StdDev |  Gen 0 | Allocated |
+|-------------------------------------------- |---------:|--------:|--------:|-------:|----------:|
+|                 UninstrumentedAspNetCoreApp | 155.6 us | 2.63 us | 2.33 us | 0.9766 |      5 KB |
+| InstrumentedAspNetCoreAppWithDefaultOptions | 176.8 us | 3.24 us | 2.70 us | 1.2207 |      7 KB |
+*/
+
+namespace Benchmarks.Instrumentation
+{
+    [InProcess]
+    public class AspNetCoreInstrumentationBenchmarks
+    {
+        private HttpClient httpClient;
+        private WebApplication app;
+        private TracerProvider tracerProvider;
+
+        [GlobalSetup(Target = nameof(UninstrumentedAspNetCoreApp))]
+        public void UninstrumentedAspNetCoreAppGlobalSetup()
+        {
+            var builder = WebApplication.CreateBuilder();
+            builder.Services.AddControllers();
+            builder.Logging.ClearProviders();
```

Comment:
do we need to?

Reply:
Yes - lot of internal logging happens while running the benchmarks (for each request).

Reply:
if we are doing the same for both instrumented and uninstrumented, we are good. Maybe its a good idea to extract this common logic to a helper method, so we know it'll be consistent.